src/Microsoft.AspNetCore.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -259,6 +271,35 @@ public override ValueTask<int> ReadAsyncImplementation(ArraySegment<byte> buffer
                         }
                     }
                 }
+                while (_mode == Mode.Trailer)
+                {
+                    // Consume request terminator
+                    var scan = input.ConsumingStart();
+                    var consumed = scan;
+                    var ch1 = scan.Take();
+                    var ch2 = scan.Take();
+                    if (ch1 == -1 || ch2 == -1)
+                    {
+                        input.ConsumingComplete(consumed, scan);
+                        if (input.RemoteIntakeFin)
+                        {
+                            _mode = Mode.Complete;
+                            return 0;
+                        }
+
+                        await input;
+                    }
+                    else if (ch1 == '\r' && ch2 == '\n')
+                    {
+                        input.ConsumingComplete(scan, scan);
+                        _mode = Mode.Complete;
+                    }
+                    else
+                    {
+                        // Post request headers
+                        throw new NotImplementedException("INVALID REQUEST FORMAT");
```

Comment:
Oh, if you're not going to process incoming trailer headers you should drain them and file a bug to deal with them later.